find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.java

```diff
@@ -0,0 +1,292 @@
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.PictureMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.tasks.route.DirectionManeuver;
+import com.esri.arcgisruntime.tasks.route.Route;
+import com.esri.arcgisruntime.tasks.route.RouteParameters;
+import com.esri.arcgisruntime.tasks.route.RouteResult;
+import com.esri.arcgisruntime.tasks.route.RouteTask;
+import com.esri.arcgisruntime.tasks.route.Stop;
+
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+
+    private static final String TAG = "FindRouteSample";
+    ArcGISVectorTiledLayer mVectorTiledLayer;
+    FloatingActionButton mDirectionFab;
+    private ProgressDialog mProgressDialog;
+
+    private MapView mMapView;
+    private RouteTask mRouteTask;
+    private RouteParameters mRouteParams;
+    private Point mSourcePoint;
+    private Point mDestinationPoint;
+    private Route mRoute;
+    private SimpleLineSymbol mRouteSymbol;
+    private GraphicsOverlay mGraphicsOverlay;
+
+    private DrawerLayout mDrawerLayout;
+    private ListView mDrawerList;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_drawer);
+
+        // inflate MapView from layout
+        mMapView = (MapView) findViewById(R.id.mapView);
+        // create new Vector Tiled Layer from service url
+        mVectorTiledLayer = new ArcGISVectorTiledLayer(
+                getResources().getString(R.string.navigation_vector));
+
+        // set tiled layer as basemap
+        Basemap basemap = new Basemap(mVectorTiledLayer);
+        // create a map with the basemap
+        ArcGISMap mMap = new ArcGISMap(basemap);
+        // create a viewpoint from lat, long, scale
+        Viewpoint sanDiegoPoint = new Viewpoint(32.7157, -117.1611, 200000);
+        // set initial map extent
+        mMap.setInitialViewpoint(sanDiegoPoint);
+        // set the map to be displayed in this view
+        mMapView.setMap(mMap);
+
+        // inflate navigation drawer
+        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
+        mDrawerList = (ListView) findViewById(R.id.left_drawer);
+
+        mDirectionFab = (FloatingActionButton) findViewById(R.id.fab);
+
+        setupDrawer();
+
+        setupSymbols();
+
+        mProgressDialog = new ProgressDialog(this);
+        mProgressDialog.setTitle(getString(R.string.progress_title));
+        mProgressDialog.setMessage(getString(R.string.progress_message));
+
+
+        mDirectionFab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                mProgressDialog.show();
+
+                getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+                getSupportActionBar().setHomeButtonEnabled(true);
```

Comment:
`getSupportActionBar)` can be null, you should check for null before calling methods.